### PR TITLE
Switch VersionHistory fetches from bulk endpoints to fetchThread & comment refresh calls

### LINKED ISSUE #1094: Switch VersionHistory fetches from bulk endpoints to fetchThread & comment refresh calls
To continue optimizing our app loading time, we should no longer fetch version history on bulkThreads or bulkOffchain, but wait until a version history modal is opened.

### PR DESCRIPTION
**Pull request by gdjohnson**
Closes #1094. DRAFT until 2/23 a.m., final round of QA desired.

## Description

To optimize our app loading time, this branch switches our fetching of thread & comment version history off of bulkThreads and bulkOffchain, waiting until either a version history modal is opened (in the case of threads) or a thread is navigated to (in the case of comments). If desired, I can further move comment version history fetching until the VH modal is opened—I refrained only because comments prefetch, on the `view_proposal` page, was already in place.

Because the UI requires a 'last edited' date in order to render a `ProposalBodyLastEdited` component (which, when clicked, opens the version history modal), this branch modifies the bulkOffchain and bulkThreads endpoints so that they parse each thread or comments' version history, grab the most recent edit's timestamp, and pass it up as `last_edited`. This value is picked up by each controller's `modelFromServer` fn, and passed as a newly added, optional `lastEdit` argument/property in the comment and thread constructors.

## How has this been tested?

I've done a round of ensuring comments & threads, both pre-existing and newly created, correctly render last edited + version history information. I want to do another round with fresh eyes tomorrow morning before opening this up fully.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [x] no

### CODE REVIEW

**Review by raykyri — CHANGES_REQUESTED** [review]
Going to "Job Board" or "First NFT Working Group call - Next Tuesday 16th Feb @ 5pm GMT" and clicking to view edits on the parent thread results in an error for me.

**Comment by gdjohnson** [comment]
OK Update... Looking at recent database entries there are some issues that need to be ironed out w/r/t UTC vs local time, and old entries not having version history. I think I'm gonna have to write a migration or two to make sure all the entries are uniform & then I can re-check this parsing code against it.

File: server/migrations/20210225201634-ensure-all-threads-have-version-history.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const query = `SELECT * FROM "OffchainThreads" WHERE version_history IS NULL OR version_history='{}'`;
+    const threads = await queryInterface.sequelize.query(query);
+    return Promise.all(threads[0].map(async (thread, i) => {
+      const firstVersion = {
+        timestamp: thread.created_at,
+        body: decodeURIComponent(thread.body)
+      };
+      const update = `UPDATE "OffchainThreads" SET version_history=ARRAY[${firstVersion}] WHERE id='${thread.id}'`;
```
Comment (raykyri) [inline server/migrations/20210225201634-ensure-all-threads-have-version-history.js:12]: Insecure SQL query, this should use a the ORM and a replacement.

File: server/migrations/20210225201634-ensure-all-threads-have-version-history.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      return queryInterface.sequelize.query(update);
+    }));
+  },
+
+  down: (queryInterface, Sequelize) => {
+    return null;
```
Comment (raykyri) [inline server/migrations/20210225201634-ensure-all-threads-have-version-history.js:18]: This won't execute. A no-op migration should return a Promise that resolves itself without doing anything.

File: client/scripts/views/pages/view_proposal/body.ts [hunk 3]
```diff
@@ -140,30 +141,46 @@ export const ProposalBodyCreated: m.Component<{
...
-export const ProposalBodyLastEdited: m.Component<{ item: AnyProposal | OffchainThread | OffchainComment<any> }> = {
+export const ProposalBodyLastEdited: m.Component<{ item: OffchainThread | OffchainComment<any> }> = {
   view: (vnode) => {
     const { item } = vnode.attrs;
     if (!item) return;
-
+    const isThread = item instanceof OffchainThread;
+    const missingVersionHistory = (!item.versionHistory || item.versionHistory.length === 0);
+    let lastEdited;
+    console.log({ lE: item.lastEdited, cA: item.createdAt });
```
Comment (raykyri) [inline client/scripts/views/pages/view_proposal/body.ts:151]: Remove

**Comment by raykyri** [comment]
Is this ready to review? It's tagged that way but I continue to get errors when testing.

**Comment by gdjohnson** [comment]
> Is this ready to review? It's tagged that way but I continue to get errors when testing.

No, sorry you went through a second QA, I haven't updated this branch since the first QA round. I've removed those A0 tags, and will try to be more diligent about removing them in the future.

File: server/migrations/20210225201634-ensure-all-threads-have-version-history.js (hunk 1 above)
Comment (gdjohnson) [reply server/migrations/20210225201634-ensure-all-threads-have-version-history.js]: Yes good to remember... Right now I'm still trying to get a workable stringified json, struggling to escape all the relevant characters in e.g. a Quill text body. This update fails for about 30% of threads, spent all day yesterday trying to debug the various issues preventing a clean data insert. 

Looking for a better high-level strategy in the meantime, unless you have a recommendation for the stringification.

Comment (gdjohnson) [reply server/migrations/20210225201634-ensure-all-threads-have-version-history.js]: (See #checkin for background on blocker)

**Comment by gdjohnson** [comment]
Alright, data's successfully migrating, I cleaned up some of the checks on the UI, marking this ready for a second review round!

**Review by raykyri — APPROVED** [review]
Seems okay. The version history modal doesn't render until the versions are fetched, which can be jarring if the amount of history retrieved is large (for Decent proposal it's 1.8MB) but that doesn't need to be fixed now.

**Pull request merged**